state/event_table_test.go

```diff
@@ -870,3 +870,78 @@ func TestRemoveUnsignedTXNID(t *testing.T) {
 		}
 	}
 }
+
+func TestEventTableSelectUnknownEventIDs(t *testing.T) {
+	db, close := connectToDB(t)
+	defer close()
+	txn, err := db.Beginx()
+	if err != nil {
+		t.Fatalf("failed to start txn: %s", err)
+	}
+	defer txn.Rollback()
+	const roomID = "!1:localhost"
+
+	// Note: there shouldn't be any other events with these IDs inserted before this
+	// transaction. $A and $B seem to be inserted and commit in TestEventTablePrevBatch.
+	const eventID1 = "$A-SelectUnknownEventIDs"
+	const eventID2 = "$B-SelectUnknownEventIDs"
+
+	knownEvents := []Event{
+		{
+			Type:     "m.room.create",
+			StateKey: "",
+			IsState:  true,
+			ID:       eventID1,
+			RoomID:   roomID,
+		},
+		{
+			Type:     "m.room.name",
+			StateKey: "",
+			IsState:  true,
+			ID:       eventID2,
+			RoomID:   roomID,
+		},
+	}
+	table := NewEventTable(db)
+
+	// Check the event IDs haven't been added by another test.
+	gotEvents, err := table.SelectByIDs(txn, true, []string{eventID1, eventID2})
+	if len(gotEvents) > 0 {
+		t.Fatalf("Event IDs already in use---commited by another test?")
+	}
+
+	// Insert the events
+	_, err = table.Insert(txn, knownEvents, false)
+	if err != nil {
+		t.Fatalf("failed to insert event: %s", err)
+	}
+
+	gotEvents, err = table.SelectByIDs(txn, true, []string{eventID1, eventID2})
+	if err != nil {
+		t.Fatalf("failed to select events: %s", err)
+	}
+	if (gotEvents[0].ID == eventID1 && gotEvents[1].ID == eventID2) || (gotEvents[0].ID == eventID2 && gotEvents[1].ID == eventID1) {
+		t.Logf("Got expected event IDs after insert. NIDS: %s=%d, %s=%d", gotEvents[0].ID, gotEvents[0].NID, gotEvents[1].ID, gotEvents[1].NID)
+	} else {
```

Comment:
So many indexes! Just do:
```go
if !reflect.DeepEqual([]string{gotEvents[0].ID, gotEvents[1].ID}, []string{eventID1, eventID2}) { ... }
```
?

Reply:
I think I was worried that the order of returned events wasn't guaranteed (like with the new query). But I see that they're ordered by NID.